rmf_site_editor/src/site/scenario.rs

```diff
@@ -18,148 +18,381 @@
+    scenarios: &Query<(Entity, &Affiliation<Entity>), With<ScenarioMarker>>,
+    instance_modifiers: &Query<(&mut InstanceModifier, &Affiliation<Entity>)>,
+) -> Option<Pose> {
+    let mut parent_pose: Option<Pose> = None;
+    let mut entity = scenario_entity;
+    while parent_pose.is_none() {
+        let Ok((_, parent_scenario)) = scenarios.get(entity) else {
+            break;
+        };
+        let Some((parent_entity, _)) = parent_scenario.0.and_then(|e| scenarios.get(e).ok()) else {
+            break;
+        };
+
+        let instance_modifier_entities =
+            get_instance_modifier_entities(parent_entity, children, instance_modifiers);
+
+        if let Some(modifier_entity) = instance_modifier_entities.get(&instance_entity) {
+            parent_pose = instance_modifiers
+                .get(*modifier_entity)
+                .ok()
+                .and_then(|(i, _)| {
+                    i.pose().or_else(|| {
+                        recall_instance
+                            .get(*modifier_entity)
+                            .ok()
+                            .and_then(|r| r.pose)
+                    })
+                });
+        }
+        entity = parent_entity;
+    }
+    parent_pose
+}
+
+/// This system searches for scenario children entities with the InstanceModifier component
+/// and maps the affiliated model instance entity to the corresponding instance modifier entity
+pub fn get_instance_modifier_entities(
+    entity: Entity,
+    children: &Query<&Children>,
+    instance_modifiers: &Query<(&mut InstanceModifier, &Affiliation<Entity>)>,
+) -> HashMap<Entity, Entity> {
+    let mut instance_to_modifier_entities = HashMap::<Entity, Entity>::new();
+    if let Ok(scenario_children) = children.get(entity) {
+        for child in scenario_children.iter() {
+            if let Some(affiliated_entity) =
+                instance_modifiers.get(*child).ok().and_then(|(_, a)| a.0)
+            {
+                instance_to_modifier_entities.insert(affiliated_entity, *child);
+            }
+        }
+    };
+    instance_to_modifier_entities
+}
+
+pub fn insert_new_instance_modifiers(
+    mut commands: Commands,
+    mut change_current_scenario: EventWriter<ChangeCurrentScenario>,
+    mut instance_modifiers: Query<(&mut InstanceModifier, &Affiliation<Entity>)>,
+    added_instances: Query<(Entity, &Pose), (Added<InstanceMarker>, Without<Pending>)>,
+    added_scenarios: Query<(Entity, &Affiliation<Entity>), Added<ScenarioMarker>>,
+    children: Query<&Children>,
+    current_scenario: Res<CurrentScenario>,
+    scenarios: Query<Entity, With<ScenarioMarker>>,
+) {
+    let Some(current_scenario_entity) = current_scenario.0 else {
+        return;
+    };
+    // Insert instance modifier entities when new scenarios are created
+    for (scenario_entity, parent_scenario) in added_scenarios.iter() {
+        if let Some(parent_scenario_entity) = parent_scenario.0 {
```

Comment:
When a new scenario is added we should give it a modifier for every instance in the site, not just for every instance that its parent has a modifier for.

With the current implementation, when you add a new root scenario, the entire list of instances in the `view_model_instances` panel is blank.